identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -40,64 +45,66 @@ public UserService(
     this.passwordEncoder = passwordEncoder;
   }
 
-  public CamundaUser createUser(final CamundaUserWithPassword userWithCredential) {
+  public CamundaUser createUserFailIfExists(
+      @Valid final CamundaUserWithPassword camundaUserWithPassword) {
     try {
-      final UserDetails userDetails =
-          User.withUsername(userWithCredential.getUsername())
-              .password(userWithCredential.getPassword())
-              .passwordEncoder(passwordEncoder::encode)
-              .disabled(!userWithCredential.isEnabled())
-              .roles("DEFAULT_USER")
-              .build();
-      camundaUserDetailsManager.createUser(userDetails);
-      final var createdUser = userProfileRepository.findByUsername(userDetails.getUsername());
-      userProfileRepository.save(
-          new Profile(
-              createdUser.getId(), userWithCredential.getEmail(), userWithCredential.getName()));
-      return userProfileRepository.findByUsername(userWithCredential.getUsername());
+      return createUser(camundaUserWithPassword);
     } catch (final DuplicateKeyException e) {
-      throw new RuntimeException("user.duplicate");
+      throw new IllegalArgumentException("User invalid.");
     }
   }
 
-  public void deleteUser(final Long id) {
+  public CamundaUser createUserIfNotExists(
```

Comment:
❓ similar question here to above, is it relevant to me that the user is created if they don't exist? I'd like to create a user here, the result from this method is the same either way, I want a user to exist :)

Reply:
That's also a way to look at it ... although I wonder about this scenario:

- user "fubar" exists in DB
- someone creates a new user "fubar" and gets an OK as if user was created
- someone thinks this is the new user they have created and wants to update the user or do stuff with that user ... **although** that user is not the actual new user but an existing one

Wouldn't that lead to potential issues?

Reply:
see above :)